src/fstab-generator/fstab-generator.c

```diff
@@ -936,6 +936,19 @@ static int parse_fstab_one(
                         mount_is_network(fstype, options) ? SPECIAL_REMOTE_FS_TARGET :
                                                             SPECIAL_LOCAL_FS_TARGET;
 
+        /* nofail or noauto don't make sense for critical filesystems we must mount in initrd. */
+        if ((is_sysroot || is_sysroot_usr)) {
+                if (FLAGS_SET(flags, MOUNT_NOFAIL) || FLAGS_SET(flags, MOUNT_NOAUTO)) {
+                        flags &= ~(MOUNT_NOFAIL|MOUNT_NOAUTO);
+                        r = fstab_filter_options(options, "noauto\0nofail", NULL, NULL, NULL, &opts);
+                        if (r < 0)
+                                return r;
+
+                        log_debug("'noauto' and 'nofail' options are ignored for /sysroot and /sysroot/usr mounts.");
```

Comment:
please suffix paths to dirs in log messages and docs with `/`. i.e. `/sysroot` → `/sysroot/` please